pkg/minikube/machine/client.go

```diff
@@ -183,7 +186,21 @@ func (api *LocalClient) Create(h *host.Host) error {
 	}{
 		{
 			"bootstrapping certificates",
-			func() error { return cert.BootstrapCertificates(h.AuthOptions()) },
+			func() error {
+				// CA cert and client cert should be generated atomically, otherwise might cause bad certificate error
+				lockErr := api.flock.LockWithTimeout(time.Second * 5)
+				if lockErr != nil {
+					return fmt.Errorf("failed to acquire bootstrap client lock: %v " + lockErr.Error())
+				}
+				defer func() {
+					lockErr = api.flock.Unlock()
+					if lockErr != nil {
+						klog.Errorf("falied to release bootstrap cert client lock  %v", lockErr.Error())
```

Comment:
colon here too

klog.Errorf("falied to release bootstrap cert client lock:  %v")